New packaging error string fails test suite

Packaging 22 changed the error message once again:

```
[   24s] 
[   24s] =================================== FAILURES ===================================
[   24s] _ test_load[\n[project]\nname = 'test'\ndependencies = [\n    'definitely not a valid PEP 508 requirement!',\n]\n-Field `project.dependencies` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "'not a va'": Expected string_end`)] _
[   24s] 
[   24s] self = <[AttributeError("'Requirement' object has no attribute 'name'") raised in repr()] Requirement object at 0x7f35674b8b80>
[   24s] requirement_string = 'definitely not a valid PEP 508 requirement!'
[   24s] 
[   24s]     def __init__(self, requirement_string: str) -> None:
[   24s]         try:
[   24s] >           parsed = parse_requirement(requirement_string)
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/requirements.py:35: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] source = 'definitely not a valid PEP 508 requirement!'
[   24s] 
[   24s]     def parse_requirement(source: str) -> ParsedRequirement:
[   24s] >       return _parse_requirement(Tokenizer(source, rules=DEFAULT_RULES))
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/_parser.py:64: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] tokenizer = <packaging._tokenizer.Tokenizer object at 0x7f35674b8f70>
[   24s] 
[   24s]     def _parse_requirement(tokenizer: Tokenizer) -> ParsedRequirement:
[   24s]         """
[   24s]         requirement = WS? IDENTIFIER WS? extras WS? requirement_details
[   24s]         """
[   24s]         tokenizer.consume("WS")
[   24s]     
[   24s]         name_token = tokenizer.expect(
[   24s]             "IDENTIFIER", expected="package name at the start of dependency specifier"
[   24s]         )
[   24s]         name = name_token.text
[   24s]         tokenizer.consume("WS")
[   24s]     
[   24s]         extras = _parse_extras(tokenizer)
[   24s]         tokenizer.consume("WS")
[   24s]     
[   24s] >       url, specifier, marker = _parse_requirement_details(tokenizer)
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/_parser.py:82: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] tokenizer = <packaging._tokenizer.Tokenizer object at 0x7f35674b8f70>
[   24s] 
[   24s]     def _parse_requirement_details(
[   24s]         tokenizer: Tokenizer,
[   24s]     ) -> Tuple[str, str, Optional[MarkerList]]:
[   24s]         """
[   24s]         requirement_details = AT URL (WS requirement_marker)?
[   24s]                             | specifier WS? (requirement_marker)?
[   24s]         """
[   24s]     
[   24s]         specifier = ""
[   24s]         url = ""
[   24s]         marker = None
[   24s]     
[   24s]         if tokenizer.check("AT"):
[   24s]             tokenizer.read()
[   24s]             tokenizer.consume("WS")
[   24s]     
[   24s]             url_start = tokenizer.position
[   24s]             url = tokenizer.expect("URL", expected="URL after @").text
[   24s]             if tokenizer.check("END", peek=True):
[   24s]                 return (url, specifier, marker)
[   24s]     
[   24s]             tokenizer.expect("WS", expected="whitespace after URL")
[   24s]     
[   24s]             marker = _parse_requirement_marker(
[   24s]                 tokenizer, span_start=url_start, after="URL and whitespace"
[   24s]             )
[   24s]         else:
[   24s]             specifier_start = tokenizer.position
[   24s]             specifier = _parse_specifier(tokenizer)
[   24s]             tokenizer.consume("WS")
[   24s]     
[   24s]             if tokenizer.check("END", peek=True):
[   24s]                 return (url, specifier, marker)
[   24s]     
[   24s] >           marker = _parse_requirement_marker(
[   24s]                 tokenizer,
[   24s]                 span_start=specifier_start,
[   24s]                 after=(
[   24s]                     "version specifier"
[   24s]                     if specifier
[   24s]                     else "name and no valid version specifier"
[   24s]                 ),
[   24s]             )
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/_parser.py:122: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] tokenizer = <packaging._tokenizer.Tokenizer object at 0x7f35674b8f70>
[   24s] 
[   24s]     def _parse_requirement_marker(
[   24s]         tokenizer: Tokenizer, *, span_start: int, after: str
[   24s]     ) -> MarkerList:
[   24s]         """
[   24s]         requirement_marker = SEMICOLON marker WS?
[   24s]         """
[   24s]     
[   24s]         if not tokenizer.check("SEMICOLON"):
[   24s] >           tokenizer.raise_syntax_error(
[   24s]                 f"Expected end or semicolon (after {after})",
[   24s]                 span_start=span_start,
[   24s]             )
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/_parser.py:143: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] self = <packaging._tokenizer.Tokenizer object at 0x7f35674b8f70>
[   24s] message = 'Expected end or semicolon (after name and no valid version specifier)'
[   24s] 
[   24s]     def raise_syntax_error(
[   24s]         self,
[   24s]         message: str,
[   24s]         *,
[   24s]         span_start: Optional[int] = None,
[   24s]         span_end: Optional[int] = None,
[   24s]     ) -> NoReturn:
[   24s]         """Raise ParserSyntaxError at the given position."""
[   24s]         span = (
[   24s]             self.position if span_start is None else span_start,
[   24s]             self.position if span_end is None else span_end,
[   24s]         )
[   24s] >       raise ParserSyntaxError(
[   24s]             message,
[   24s]             source=self.source,
[   24s]             span=span,
[   24s]         )
[   24s] E       packaging._tokenizer.ParserSyntaxError: Expected end or semicolon (after name and no valid version specifier)
[   24s] E           definitely not a valid PEP 508 requirement!
[   24s] E                      ^
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/_tokenizer.py:161: ParserSyntaxError
[   24s] 
[   24s] The above exception was the direct cause of the following exception:
[   24s] 
[   24s] fetcher = <pyproject_metadata.DataFetcher object at 0x7f35674b8730>
[   24s] 
[   24s]     @staticmethod
[   24s]     def _get_dependencies(fetcher: DataFetcher) -> List[packaging.requirements.Requirement]:
[   24s]         try:
[   24s]             requirement_strings = fetcher.get_list('project.dependencies')
[   24s]         except KeyError:
[   24s]             return []
[   24s]     
[   24s]         requirements: List[packaging.requirements.Requirement] = []
[   24s]         for req in requirement_strings:
[   24s]             try:
[   24s] >               requirements.append(packaging.requirements.Requirement(req))
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/pyproject_metadata/__init__.py:449: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] self = <[AttributeError("'Requirement' object has no attribute 'name'") raised in repr()] Requirement object at 0x7f35674b8b80>
[   24s] requirement_string = 'definitely not a valid PEP 508 requirement!'
[   24s] 
[   24s]     def __init__(self, requirement_string: str) -> None:
[   24s]         try:
[   24s]             parsed = parse_requirement(requirement_string)
[   24s]         except ParserSyntaxError as e:
[   24s] >           raise InvalidRequirement(str(e)) from e
[   24s] E           packaging.requirements.InvalidRequirement: Expected end or semicolon (after name and no valid version specifier)
[   24s] E               definitely not a valid PEP 508 requirement!
[   24s] E                          ^
[   24s] 
[   24s] /usr/lib/python3.8/site-packages/packaging/requirements.py:37: InvalidRequirement
[   24s] 
[   24s] During handling of the above exception, another exception occurred:
[   24s] 
[   24s] package = PosixPath('/home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/packages/full-metadata')
[   24s] data = "\n[project]\nname = 'test'\ndependencies = [\n    'definitely not a valid PEP 508 requirement!',\n]\n"
[   24s] error = 'Field `project.dependencies` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "\'not a va\'": Expected string_end`)'
[   24s] 
[   24s]     @pytest.mark.parametrize(
...
[   24s]             (
[   24s]                 textwrap.dedent('''
[   24s]                     [project]
[   24s]                     name = 'test'
[   24s]                     dependencies = [
[   24s]                         'definitely not a valid PEP 508 requirement!',
[   24s]                     ]
[   24s]                 '''),
[   24s]                 (
[   24s]                     'Field `project.dependencies` contains an invalid PEP 508 requirement '
[   24s]                     'string `definitely not a valid PEP 508 requirement!` '
[   24s]                     '(`Parse error at "\'not a va\'": Expected string_end`)'
[   24s]                 ),
[   24s]             ),
...
[   24s]         ]
[   24s]     )
[   24s]     def test_load(package, data, error):
[   24s]         with pytest.raises(pyproject_metadata.ConfigurationError, match=re.escape(error)):
[   24s] >           pyproject_metadata.StandardMetadata.from_pyproject(tomli.loads(data))
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/test_standard_metadata.py:481: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] cls = <class 'pyproject_metadata.StandardMetadata'>
[   24s] data = {'project': {'dependencies': ['definitely not a valid PEP 508 requirement!'], 'name': 'test'}}
[   24s] project_dir = PosixPath('.')
[   24s] 
[   24s]     @classmethod
[   24s]     def from_pyproject(
[   24s]         cls,
[   24s]         data: Mapping[str, Any],
[   24s]         project_dir: Union[str, os.PathLike[str]] = os.path.curdir,
[   24s]     ) -> StandardMetadata:
[   24s]         fetcher = DataFetcher(data)
[   24s]         project_dir = pathlib.Path(project_dir)
[   24s]     
[   24s]         if 'project' not in fetcher:
[   24s]             raise ConfigurationError('Section `project` missing in pyproject.toml')
[   24s]     
[   24s]         dynamic = fetcher.get_list('project.dynamic')
[   24s]         if 'name' in dynamic:
[   24s]             raise ConfigurationError('Unsupported field `name` in `project.dynamic`')
[   24s]     
[   24s]         for field in dynamic:
[   24s]             if field in data['project']:
[   24s]                 raise ConfigurationError(
[   24s]                     f'Field `project.{field}` declared as dynamic in but is defined'
[   24s]                 )
[   24s]     
[   24s]         name = fetcher.get_str('project.name')
[   24s]         if not name:
[   24s]             raise ConfigurationError('Field `project.name` missing')
[   24s]     
[   24s]         version_string = fetcher.get_str('project.version')
[   24s]         requires_python_string = fetcher.get_str('project.requires-python')
[   24s]     
[   24s]         return cls(
[   24s]             name,
[   24s]             packaging.version.Version(version_string) if version_string else None,
[   24s]             fetcher.get_str('project.description'),
[   24s]             cls._get_license(fetcher, project_dir),
[   24s]             cls._get_readme(fetcher, project_dir),
[   24s]             packaging.specifiers.SpecifierSet(requires_python_string) if requires_python_string else None,
[   24s] >           cls._get_dependencies(fetcher),
[   24s]             cls._get_optional_dependencies(fetcher),
[   24s]             cls._get_entrypoints(fetcher),
[   24s]             fetcher.get_people('project.authors'),
[   24s]             fetcher.get_people('project.maintainers'),
[   24s]             fetcher.get_dict('project.urls'),
[   24s]             fetcher.get_list('project.classifiers'),
[   24s]             fetcher.get_list('project.keywords'),
[   24s]             fetcher.get_dict('project.scripts'),
[   24s]             fetcher.get_dict('project.gui-scripts'),
[   24s]             dynamic,
[   24s]         )
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/pyproject_metadata/__init__.py:238: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] fetcher = <pyproject_metadata.DataFetcher object at 0x7f35674b8730>
[   24s] 
[   24s]     @staticmethod
[   24s]     def _get_dependencies(fetcher: DataFetcher) -> List[packaging.requirements.Requirement]:
[   24s]         try:
[   24s]             requirement_strings = fetcher.get_list('project.dependencies')
[   24s]         except KeyError:
[   24s]             return []
[   24s]     
[   24s]         requirements: List[packaging.requirements.Requirement] = []
[   24s]         for req in requirement_strings:
[   24s]             try:
[   24s]                 requirements.append(packaging.requirements.Requirement(req))
[   24s]             except packaging.requirements.InvalidRequirement as e:
[   24s] >               raise ConfigurationError(
[   24s]                     'Field `project.dependencies` contains an invalid PEP 508 '
[   24s]                     f'requirement string `{req}` (`{str(e)}`)'
[   24s]                 )
[   24s] E               pyproject_metadata.ConfigurationError: Field `project.dependencies` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Expected end or semicolon (after name and no valid version specifier)
[   24s] E                   definitely not a valid PEP 508 requirement!
[   24s] E                              ^`)
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/pyproject_metadata/__init__.py:451: ConfigurationError
[   24s] 
[   24s] During handling of the above exception, another exception occurred:
[   24s] 
[   24s] package = PosixPath('/home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/packages/full-metadata')
[   24s] data = "\n[project]\nname = 'test'\ndependencies = [\n    'definitely not a valid PEP 508 requirement!',\n]\n"
[   24s] error = 'Field `project.dependencies` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "\'not a va\'": Expected string_end`)'
[   24s] 
[   24s]     @pytest.mark.parametrize(
[   24s]         ('data', 'error'),
[   24s]         [
[   24s]                 textwrap.dedent('''
[   24s]                     [project]
[   24s]                     name = 'test'
[   24s]                     dependencies = [
[   24s]                         'definitely not a valid PEP 508 requirement!',
[   24s]                     ]
[   24s]                 '''),
[   24s]                 (
[   24s]                     'Field `project.dependencies` contains an invalid PEP 508 requirement '
[   24s]                     'string `definitely not a valid PEP 508 requirement!` '
[   24s]                     '(`Parse error at "\'not a va\'": Expected string_end`)'
[   24s]                 ),
[   24s]             (
[   24s]                 textwrap.dedent('''

[   24s]     def test_load(package, data, error):
[   24s]         with pytest.raises(pyproject_metadata.ConfigurationError, match=re.escape(error)):
[   24s] >           pyproject_metadata.StandardMetadata.from_pyproject(tomli.loads(data))
[   24s] E           AssertionError: Regex pattern 'Field\\ `project\\.dependencies`\\ contains\\ an\\ invalid\\ PEP\\ 508\\ requirement\\ string\\ `definitely\\ not\\ a\\ valid\\ PEP\\ 508\\ requirement!`\\ \\(`Parse\\ error\\ at\\ "\'not\\ a\\ va\'":\\ Expected\\ string_end`\\)' does not match 'Field `project.dependencies` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Expected end or semicolon (after name and no valid version specifier)\n    definitely not a valid PEP 508 requirement!\n               ^`)'.
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/test_standard_metadata.py:481: AssertionError
[   24s] _ test_load[\n[project]\nname = 'test'\n[project.optional-dependencies]\ntest = [\n    'definitely not a valid PEP 508 requirement!',\n]\n-Field `project.optional-dependencies.test` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "'not a va'": Expected string_end`)] _
...
[   24s]     )
[   24s]     def test_load(package, data, error):
[   24s]         with pytest.raises(pyproject_metadata.ConfigurationError, match=re.escape(error)):
[   24s] >           pyproject_metadata.StandardMetadata.from_pyproject(tomli.loads(data))
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/test_standard_metadata.py:481: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] cls = <class 'pyproject_metadata.StandardMetadata'>
[   24s] data = {'project': {'name': 'test', 'optional-dependencies': {'test': ['definitely not a valid PEP 508 requirement!']}}}
[   24s] project_dir = PosixPath('.')
[   24s] 
[   24s]     @classmethod
[   24s]     def from_pyproject(
[   24s]         cls,
[   24s]         data: Mapping[str, Any],
[   24s]         project_dir: Union[str, os.PathLike[str]] = os.path.curdir,
[   24s]     ) -> StandardMetadata:
[   24s]         fetcher = DataFetcher(data)
[   24s]         project_dir = pathlib.Path(project_dir)
[   24s]     
[   24s]         if 'project' not in fetcher:
[   24s]             raise ConfigurationError('Section `project` missing in pyproject.toml')
[   24s]     
[   24s]         dynamic = fetcher.get_list('project.dynamic')
[   24s]         if 'name' in dynamic:
[   24s]             raise ConfigurationError('Unsupported field `name` in `project.dynamic`')
[   24s]     
[   24s]         for field in dynamic:
[   24s]             if field in data['project']:
[   24s]                 raise ConfigurationError(
[   24s]                     f'Field `project.{field}` declared as dynamic in but is defined'
[   24s]                 )
[   24s]     
[   24s]         name = fetcher.get_str('project.name')
[   24s]         if not name:
[   24s]             raise ConfigurationError('Field `project.name` missing')
[   24s]     
[   24s]         version_string = fetcher.get_str('project.version')
[   24s]         requires_python_string = fetcher.get_str('project.requires-python')
[   24s]     
[   24s]         return cls(
[   24s]             name,
[   24s]             packaging.version.Version(version_string) if version_string else None,
[   24s]             fetcher.get_str('project.description'),
[   24s]             cls._get_license(fetcher, project_dir),
[   24s]             cls._get_readme(fetcher, project_dir),
[   24s]             packaging.specifiers.SpecifierSet(requires_python_string) if requires_python_string else None,
[   24s]             cls._get_dependencies(fetcher),
[   24s] >           cls._get_optional_dependencies(fetcher),
[   24s]             cls._get_entrypoints(fetcher),
[   24s]             fetcher.get_people('project.authors'),
[   24s]             fetcher.get_people('project.maintainers'),
[   24s]             fetcher.get_dict('project.urls'),
[   24s]             fetcher.get_list('project.classifiers'),
[   24s]             fetcher.get_list('project.keywords'),
[   24s]             fetcher.get_dict('project.scripts'),
[   24s]             fetcher.get_dict('project.gui-scripts'),
[   24s]             dynamic,
[   24s]         )
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/pyproject_metadata/__init__.py:239: 
[   24s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   24s] 
[   24s] fetcher = <pyproject_metadata.DataFetcher object at 0x7f35672949a0>
[   24s] 
[   24s]     @staticmethod
[   24s]     def _get_optional_dependencies(fetcher: DataFetcher) -> Dict[str, List[packaging.requirements.Requirement]]:
[   24s]         try:
[   24s]             val = fetcher.get('project.optional-dependencies')
[   24s]         except KeyError:
[   24s]             return {}
[   24s]     
[   24s]         requirements_dict: DefaultDict[str, List[packaging.requirements.Requirement]] = collections.defaultdict(list)
[   24s]         if not isinstance(val, dict):
[   24s]             raise ConfigurationError(
[   24s]                 'Field `project.optional-dependencies` has an invalid type, expecting a '
[   24s]                 f'dictionary of PEP 508 requirement strings (got `{val}`)'
[   24s]             )
[   24s]         for extra, requirements in val.copy().items():
[   24s]             assert isinstance(extra, str)
[   24s]             if not isinstance(requirements, list):
[   24s]                 raise ConfigurationError(
[   24s]                     f'Field `project.optional-dependencies.{extra}` has an invalid type, expecting a '
[   24s]                     f'dictionary PEP 508 requirement strings (got `{requirements}`)'
[   24s]                 )
[   24s]             for i, req in enumerate(requirements):
[   24s]                 if not isinstance(req, str):
[   24s]                     raise ConfigurationError(
[   24s]                         f'Field `project.optional-dependencies.{extra}` has an invalid type, '
[   24s]                         f'expecting a PEP 508 requirement string (got `{req}`)'
[   24s]                     )
[   24s]                 try:
[   24s]                     requirements_dict[extra].append(packaging.requirements.Requirement(req))
[   24s]                 except packaging.requirements.InvalidRequirement as e:
[   24s] >                   raise ConfigurationError(
[   24s]                         f'Field `project.optional-dependencies.{extra}` contains '
[   24s]                         f'an invalid PEP 508 requirement string `{req}` (`{str(e)}`)'
[   24s]                     )
[   24s] E                   pyproject_metadata.ConfigurationError: Field `project.optional-dependencies.test` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Expected end or semicolon (after name and no valid version specifier)
[   24s] E                       definitely not a valid PEP 508 requirement!
[   24s] E                                  ^`)
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/pyproject_metadata/__init__.py:486: ConfigurationError
[   24s] 
[   24s] During handling of the above exception, another exception occurred:
[   24s] 
[   24s] package = PosixPath('/home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/packages/full-metadata')
[   24s] data = "\n[project]\nname = 'test'\n[project.optional-dependencies]\ntest = [\n    'definitely not a valid PEP 508 requirement!',\n]\n"
[   24s] error = 'Field `project.optional-dependencies.test` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "\'not a va\'": Expected string_end`)'
[   24s] 
[   24s]     @pytest.mark.parametrize(
[   24s]         ('data', 'error'),
...
[   24s]         ]
[   24s]     )
[   24s]     def test_load(package, data, error):
[   24s]         with pytest.raises(pyproject_metadata.ConfigurationError, match=re.escape(error)):
[   24s] >           pyproject_metadata.StandardMetadata.from_pyproject(tomli.loads(data))
[   24s] E           AssertionError: Regex pattern 'Field\\ `project\\.optional\\-dependencies\\.test`\\ contains\\ an\\ invalid\\ PEP\\ 508\\ requirement\\ string\\ `definitely\\ not\\ a\\ valid\\ PEP\\ 508\\ requirement!`\\ \\(`Parse\\ error\\ at\\ "\'not\\ a\\ va\'":\\ Expected\\ string_end`\\)' does not match 'Field `project.optional-dependencies.test` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Expected end or semicolon (after name and no valid version specifier)\n    definitely not a valid PEP 508 requirement!\n               ^`)'.
[   24s] 
[   24s] /home/abuild/rpmbuild/BUILD/python-pyproject-metadata-0.6.1/tests/test_standard_metadata.py:481: AssertionError
[   24s] =========================== short test summary info ============================
[   24s] FAILED tests/test_standard_metadata.py::test_load[\n[project]\nname = 'test'\ndependencies = [\n    'definitely not a valid PEP 508 requirement!',\n]\n-Field `project.dependencies` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "'not a va'": Expected string_end`)]
[   24s] FAILED tests/test_standard_metadata.py::test_load[\n[project]\nname = 'test'\n[project.optional-dependencies]\ntest = [\n    'definitely not a valid PEP 508 requirement!',\n]\n-Field `project.optional-dependencies.test` contains an invalid PEP 508 requirement string `definitely not a valid PEP 508 requirement!` (`Parse error at "'not a va'": Expected string_end`)]
[   24s] ========================= 2 failed, 69 passed in 0.60s =========================
[   24s] error: Bad exit status from /var/tmp/rpm-tmp.xeWQjV (%check)
```

--- a/setup.cfg
+++ b/setup.cfg
@@ -23,7 +23,7 @@ project_urls =
 packages = find:
 include_package_data = True
 install_requires =
-    packaging>=19.0
+    packaging>=22.0
 python_requires = >=3.7
 
 [options.packages.find]

--- a/tests/test_standard_metadata.py
+++ b/tests/test_standard_metadata.py
@@ -236,7 +236,7 @@ from .conftest import cd_package
             (
                 'Field `project.dependencies` contains an invalid PEP 508 requirement '
                 'string `definitely not a valid PEP 508 requirement!` '
-                '(`Parse error at "\'not a va\'": Expected string_end`)'
+                '(`Expected end or semicolon (after name and no valid version specifier)'
             ),
         ),
         # optional-dependencies
@@ -289,7 +289,7 @@ from .conftest import cd_package
             (
                 'Field `project.optional-dependencies.test` contains an invalid '
                 'PEP 508 requirement string `definitely not a valid PEP 508 requirement!` '
-                '(`Parse error at "\'not a va\'": Expected string_end`)'
+                '(`Expected end or semicolon (after name and no valid version specifier)'
             ),
         ),
         # requires-python

